mlrun/alerts/alert.py

```diff
@@ -0,0 +1,133 @@
+        reset_policy: alert_constants.ResetPolicy = None,
+        notifications: list[Notification] = None,
+        entity: alert_constants.EventEntity = None,
+        id: int = None,
+        state: alert_constants.AlertActiveState = None,
+        created: str = None,
+        count: int = None,
+    ):
+        self.project = project
+        self.name = name
+        self.description = description
+        self.summary = summary
+        self.severity = severity
+        self.trigger = trigger
+        self.criteria = criteria
+        self.reset_policy = reset_policy
+        self.notifications = notifications or []
+        self.entity = entity
+        self.id = id
+        self.state = state
+        self.created = created
+        self.count = count
+
+        if template:
+            self._apply_template(template)
+
+    def to_dict(self, fields: list = None, exclude: list = None, strip: bool = False):
+        data = super().to_dict(self._dict_fields)
+
+        data["entity"] = (
+            self.entity.dict() if not isinstance(self.entity, dict) else self.entity
+        )
+        data["notifications"] = [
+            notification.dict() if not isinstance(notification, dict) else notification
+            for notification in self.notifications
+        ]
+        data["trigger"] = (
+            self.trigger.dict() if not isinstance(self.trigger, dict) else self.trigger
+        )
+        return data
+
+    def from_dict(self, struct=None, fields=None, deprecated_fields: dict = None):
+        new_obj = super().from_dict(struct, self._dict_fields)
+
+        entity_data = struct.get("entity")
+        if entity_data:
+            entity_obj = alert_constants.EventEntity.parse_obj(entity_data)
+            new_obj.entity = entity_obj
+
+        notifications_data = struct.get("notifications")
+        if notifications_data:
+            notifications_objs = [
+                Notification.parse_obj(notification_data)
+                for notification_data in notifications_data
+            ]
+            new_obj.notifications = notifications_objs
+
+        trigger_data = struct.get("trigger")
+        if trigger_data:
+            trigger_obj = alert_constants.AlertTrigger.parse_obj(trigger_data)
+            new_obj.trigger = trigger_obj
+
+        return new_obj
+
+    def with_notifications(self, notifications: list[Notification]):
+        if not isinstance(notifications, list) or not all(
+            isinstance(item, Notification) for item in notifications
+        ):
+            raise ValueError("Notifications parameter must be a list of notifications")
+        for notification in notifications:
+            self.notifications.append(notification)
+        return self
+
+    def with_entity(self, entity: alert_constants.EventEntity):
+        if not isinstance(entity, alert_constants.EventEntity):
+            raise ValueError("entity parameter must be of type: EventEntity")
```

Comment:
capitalize the errors
```suggestion
            raise ValueError("Entity parameter must be of type: EventEntity")
```